Sources/PrimerSDK/Classes/User Interface/Root/PrimerRootViewController.swift

```diff
@@ -532,15 +490,36 @@ extension PrimerRootViewController: UIGestureRecognizerDelegate {
     }
 }
 
+
+extension PrimerRootViewController {
+    
+    private func handleErrorBasedOnSDKSettings(_ error: Error) {
+        
+        let settings: PrimerSettingsProtocol = DependencyContainer.resolve()
+        if settings.isManualPaymentHandlingEnabled {
+            PrimerDelegateProxy.onResumeError(error)
+            handle(error: error)
+        } else {
+            Primer.shared.delegate?.primerDidFailWithError?(error, data: nil, decisionHandler: { errorDecision in
```

Comment:
Call delegate functions via the proxy.

Reply:
Within the proxy make a check whether the delegate function is implemented, if it's not continue the flow (success or error with an empty message).

Reply:
Addressed ✅